analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsShouldBeLogged.Microsoft.Extensions.Logging.cs

```diff
@@ -0,0 +1,405 @@
+        {
+            logger.LogError(new EventId(1), e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogError("Message!"); // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogInformation(e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogInformation("Message!"); // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogInformation(new EventId(1), e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogInformation("Message!"); // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogTrace(e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogTrace("Message!"); // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogTrace(new EventId(1), e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogTrace("Message!"); // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning(e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning("Message!"); // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning("Message!"); // Noncompliant
+        }
+    }
+
+    public void MultipleLogsInTheSameCatch()
+    {
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+            logger.LogWarning(new EventId(1), e, "Message!");
+        }
+
+        try { }
+        catch (DivideByZeroException e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+            logger.LogInformation(new EventId(1), "Message!");          // Compliant - the exception has been loged already
+        }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+            if (true)
+            {
+                logger.LogInformation(new EventId(1), "Message!");      // Compliant - the exception has been loged already
+            }
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), "Message!"); // Noncompliant
+            logger.LogWarning(new EventId(1), "Message!"); // Noncompliant
+        }
+    }
+
+    public void UsageAsStaticMethod()
+    {
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.Log(logger, LogLevel.Critical, "Message!");    // Noncompliant
```

Comment:
To keep it more concise I'd only test 1 or 2 of these, and assume if they work then so does the rest.

Reply:
moved in the parameterized test